src/metabase/driver/mysql.clj

```diff
@@ -368,42 +367,20 @@
       (hsql/call :convert_tz expr (or source-timezone (qp.timezone/results-timezone-id)) target-timezone)
       "datetime")))
 
-(defn- datetime-diff-helper [x y unit]
-  (case unit
-    (:year :month)
-    (hsql/call :timestampdiff (hsql/raw (name unit)) (hsql/call :date x) (hsql/call :date y))
-
-    :week
-    (let [positive-diff (fn [a b] (hx/floor (hx// (hsql/call :datediff b a) 7)))]
-      (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
-
-    :quarter
-    (let [positive-diff (fn [a b] (hx/floor (hx// (datetime-diff-helper a b :month) 3)))]
-      (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
```

Comment:
There was no need for any of this `positive-diff` and `floor` stuff for weeks and quarters, since `timestampdiff` works out of the box